### ISSUE TITLE
Fake players linger after switching back to survival

### ISSUE THREAD

**Issue by LucasVon** [North-West-Wind/FabricSurvivalSpectator#2]
Hello! There seems to be an issue where the fake player created doesn't get removed after switching back to survival mode. Even after disconnecting and restarting the client, the fake players are still there taking up spots.

![2021-06-13_06 15 10](https://user-images.githubusercontent.com/9199524/121796171-0f684700-cc0f-11eb-8631-a6ac4a1dbe93.png)

**Comment by North-West-Wind** [comment]
Can you send me your mod list?

**Comment by LucasVon** [comment]
Sure thing. Survival Spectator is installed on the server.

Server mods:

![BlCzgq6](https://user-images.githubusercontent.com/9199524/121801491-75fd5d00-cc2f-11eb-8ca8-9d3468386530.png)

Client mods:

![iXyjuYQ](https://user-images.githubusercontent.com/9199524/121801464-48b0af00-cc2f-11eb-9dfc-51334fff2eb3.png)

**Comment by North-West-Wind** [comment]
I cannot reproduce the glitch.
When did this start happening?

EDIT: Although I cannot reproduce it, I still have an idea to fix it, but knowing exactly what happen will prevent similar glitch from happening in the future

**Comment by LucasVon** [comment]
I installed the new 2.0.2 but the issue still persists. I installed 2.0.0 earlier today and got the same "unexpected error" from the previous issue. About half an hour after you released 2.0.1 I downloaded it and replaced the old one on the server, and the glitch started right from the get go.

I tested it out on singleplayer mode and the glitch happens. I tried disabling all mods on my client except Survival Spectator, also still happens.

**Comment by aschoettler** [comment]
I am also experiencing this issue. Each time I use `/s` twice to go to spectator and then back into survival, there's an extra copy of my name appearing on the tab list.
Server mods:
```
FabricProxy-Lite-1.1.4.jar
carpet-extra-1.17-1.4.40.jar
environmentalcreepers-fabric-1.16.4-1.6.1.jar
fabric-api-0.34.10+1.17.jar
fabric-carpet-1.17-1.4.40+v210608.jar
servux-fabric-1.17.0-0.1.0.jar
shulkerboxtooltip-2.3.5+1.16.5.jar
survivalspectator-2.0.2.jar
```

**Comment by niceduckdev** [comment]
Hello there!

I am still having this issue with the latest build. I have tried porting over the 1.16 version to 1.17 myself but am unable to do so succesfully.

If you could update the mod and fix the bug or reach out to me/help me fix it I would be forever thankful.

Greetings,
niceduck

**Comment by North-West-Wind** [comment]
Fixed with v2.0.3
The update for the 1.16 version is coming later. Stay tuned!
